### PR TITLE
feat: added nonlinear param to buildFunctionFromTable

### PR DESCRIPTION
**Pull request by Ouwen**
### Context

In order to support sigmoid LUT based on dicom standard, non-linear LUT's need to be created on the fly any time a user changes the WW/WL. The current buildFunctionFromTable only support linearly spaced x input which is inefficient sampling for sigmoid functions. This change simply allows users to build functions from the long format as used in RGBLong.

### Results

This change simply allows users to build functions from the long format as used in RGBLong.

### Changes

- [x] Documentation and TypeScript definitions were updated to match those changes

### PR and Code Checklist

- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing

Because this is a surgical change, no new tests have been added.

### CODE REVIEW

**Comment by Ouwen** [comment]
@floryst hope you are doing well, this is my first time contributing to vtkjs so please let me know if there are any additional modifications I need to make!

File: Sources/Rendering/Core/ColorTransferFunction/index.js [hunk 1]
```diff
@@ -871,6 +871,21 @@ function vtkColorTransferFunction(publicAPI, model) {
...
+        r: table[i * 6 + 1],
+        g: table[i * 6 + 2],
+        b: table[i * 6 + 3],
+        midpoint: table[i * 6 + 4],
+        sharpness: table[i * 6 + 5],
+      };
+      model.nodes.push(node);
+    }
+    publicAPI.sortAndUpdateRange();
+  };
```
Comment (finetjul) [inline Sources/Rendering/Core/ColorTransferFunction/index.js:888]: If `table` was a `vtkDataArray`, we could get the number of components per tuple:
- 3 ->RGB
- 4 ->XRGB
- 5 -> RGBMS
- 6 -> XRGBMS

You would then have something very generic.
(here is how you could convert `table` into a `vtkDataArray`:
```
ctf.buildFunctionFromArray(vtkDataArray.newInstance({values: table, numberOfComponents: 6})
```

File: Sources/Rendering/Core/ColorTransferFunction/index.d.ts [hunk 2]
```diff
@@ -257,6 +267,11 @@ export interface vtkColorTransferFunction extends vtkObject {
    */
   setMappingRange(min: number, max: number): void;
 
+  /**
+   * Get the range of mapped scalars.
+   */
+  getRange(): number[];
```
Comment (finetjul) [inline Sources/Rendering/Core/ColorTransferFunction/index.d.ts:273]: should be defined in superclass vtkScalarsToColors

File: Sources/Rendering/Core/ColorTransferFunction/index.js (hunk 1 above)
Comment (Ouwen) [reply Sources/Rendering/Core/ColorTransferFunction/index.js:888]: makes sense, I will make that change

File: Sources/Rendering/Core/ColorTransferFunction/index.d.ts (hunk 2 above)
Comment (finetjul) [reply Sources/Rendering/Core/ColorTransferFunction/index.d.ts:273]: It should be fixed in vtkScalarsToColors, the `min` and `max` parameters are ignored by the function. 

I think they should be removed from `ScalarsToColors/index.d.ts` but also from `ScalarsToColors/index.js` (in a different commit, same PR)

**Comment by Ouwen** [comment]
@finetjul ready for your review

File: Sources/Rendering/Core/ColorTransferFunction/index.d.ts [hunk 3]
```diff
@@ -201,6 +202,14 @@ export interface vtkColorTransferFunction extends vtkObject {
     withAlpha: boolean
   ): Float32Array;
 
+  /**
+   * Construct a color transfer function from a vtkDataArray.
+   * @param {vtkDataArray} array
+   */
```
Comment (finetjul) [inline Sources/Rendering/Core/ColorTransferFunction/index.d.ts:208]: Would be nice to document how tuple size is used by the function

**Review by finetjul — APPROVED** [review]
Looking good. I have just a minor final comment regarding documentation.

File: Sources/Rendering/Core/ColorTransferFunction/index.d.ts (hunk 3 above)
Comment (Ouwen) [reply Sources/Rendering/Core/ColorTransferFunction/index.d.ts:214]: The documentation has been moved here

Comment (Ouwen) [reply Sources/Rendering/Core/ColorTransferFunction/index.d.ts:219]: good catch!

**Comment by finetjul** [comment]
Looking good.
Final comment: can you please squash your commits in a unique commit, e.g. `"feat(colortransferfunction): add buildFunctionFromArray()"`

**Comment by Ouwen** [comment]
Squashed the commits and pushed

**Review by floryst — APPROVED** [review]
LGTM. Thanks, @Ouwen!

**Comment by finetjul** [comment]
LGTM

**Comment by floryst** [comment]
@Ouwen can you rebase onto master? I'm given no options to update your branch prior to merging, so having you do that would allow us to merge.

**Comment by Ouwen** [comment]
Sure I’ll rebase to master and force push

**Comment by Ouwen** [comment]
@floryst branch is now rebased with master

**Pull request merged**